src/v/cluster/controller.cc

```diff
@@ -276,6 +276,17 @@ ss::future<> controller::start() {
             std::ref(_connections),
             std::ref(_partition_manager),
             std::ref(_raft_manager),
+            [this](
+              node_health_report const& report,
+              boost::optional<node_health_report const&> old_report) {
+                if (
+                  !old_report
+                  || report.logical_version
+                       != old_report.value().logical_version) {
+                    (void)_feature_frontend.local().update_node_version(
```

Comment:
gate? restructuring to do some of this work in a background fiber might also be reasonable

Reply:
Updated this to make the health monitor take an async callback and wait for it -- that way the health monitor's gate covers it.